### ISSUE TITLE
local variable 'new_json' referenced before assignment

### ISSUE THREAD

**Issue by Anthchirp** [xia2/xia2#273]
Interesting crash on current release candidate. Code looks similar on master.
```
XIA2 0.5.765-g530b580e-remotes/origin/dials-next
DIALS 1.12.18-g1657e4f1b
CCP4 7.0.065
Command line: xia2 pipeline=dials nproc=1 njob=2 mode=parallel trust_beam_centre=True xinfo=/scratch/jenkins_agent/workspace/dials_next_tests/build_dials/modules/pytest/popen-gw7/test_dials_split0/split.xinfo


easy_mp crash detected; subprocess trace: ----
Traceback (most recent call last):
  File "/scratch/jenkins_agent/workspace/dials_next_tests/build_dials/modules/cctbx_project/libtbx/scheduling/job_scheduler.py", line 64, in job_cycle
    value = target( *args, **kwargs )
  File "/scratch/jenkins_agent/workspace/dials_next_tests/build_dials/modules/xia2/Applications/xia2_helpers.py", line 91, in process_one_sweep
    xinfo = XProject.from_json(new_json)
local variable 'new_json' referenced before assignment
----------------------------------------------
```

**Comment by Anthchirp** [comment]
Code suggests file ```xia2_json``` does not exist and ```success == True```.
https://github.com/xia2/xia2/blob/f61016eee438ae143a7e80d2958160f5a4ff44f0/Applications/xia2_helpers.py#L81-L89

**Comment by Anthchirp** [comment]
Best guess is that both subprocesses died for an unidentified reason. Added a patch to keep the tail end of the output in ```xia2-debug.txt```. Should be helpful if this shows up again.
A crash may have been accompanied with a 0 exit code, fixed that in 2821d88fb3b8453ce50053c1b7b00f0bbbe08d2e.

Let's see if this is a fluke or if it shows up again.
